elastic/kibana
# [Infra UI] Allow selection and updating of queries by hosts

## Summary

This PR adds an option to add a filter by selecting hosts from the table. It is using a combined filter (created using this [new helper](https://github.com/elastic/kibana/pull/159968/files#diff-d16aa8d5fd369ac51f139230cabc1b5ea1f2920eefa7e6d5c9068e0a12cbaa78R11) ) to create the same filter as the lens charts. Once the filter is added the selection is removed from the table.
The final approved button design:
- Multiple selected: <img width="2140" alt="image" src="https://github.com/elastic/kibana/assets/14139027/b820a840-0926-48c1-ac40-a3e0ac310794">
- Single selection:  <img width="337" alt="image" src="https://github.com/elastic/kibana/assets/14139027/93502bbe-e695-4da6-bd36-94f3e8dcb84c">


## Testing
- Go to host view
- A combined filter should be added to the unified search bar:
   - Select all hosts and add a filter
   - Select 1 host and add a filter
   - Select multiple hosts and add a filter
- Should also "stack" multiple filters if there are existing filters coming from the lens charts/search bar

https://github.com/elastic/kibana/assets/14139027/64243fd8-822b-48a9-a8de-28bbb7ea8658
